org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.xtend

```diff
@@ -40,14 +41,14 @@ class CodeActionService implements ICodeActionService {
 
 	override getCodeActions(Document document, XtextResource resource, CodeActionParams params,
 		CancelIndicator indicator) {
-		val commands = <Command>newArrayList
+		val actions = newArrayList
 		for (d : params.context.diagnostics) {
 			switch d.code {
-				case INVALID_NAME: commands += d.fixInvalidName(document, resource, params)
-				case UNSORTED_MEMBERS: commands += d.fixUnsortedMembers(document, resource, params)
+				case INVALID_NAME: actions += Either.forLeft(d.fixInvalidName(document, resource, params))
+				case UNSORTED_MEMBERS: actions += Either.forRight(d.fixUnsortedMembers(document, resource, params))
 			}
 		}
-		return commands.map[Either.forLeft(it)];
+		return actions
 	}
 
 	def private Command fixInvalidName(Diagnostic d, Document doc, XtextResource res, CodeActionParams params) {
```

Comment:
Is there a reason you translated only one of these two methods to use code actions?

Reply:
Yes, to keep both tested